vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinServletContextInitializer.java

```diff
@@ -294,9 +296,16 @@ public void failFastContextInitialized(ServletContextEvent event) {
             Stream<Class<? extends Component>> hasErrorComponents = findBySuperType(
                     getErrorParameterPackages(), HasErrorParameter.class)
                             .filter(Component.class::isAssignableFrom)
+                            // Replace Flow default with custom version for Spring
+                            .filter(clazz -> clazz != RouteNotFoundError.class)
```

Comment:
We could probably combine those two `filter` calls into one call. I can't recall if those two filter calls lead to items traverse running twice, but at least it won't create extra Stream object.